pkg/controller/cluster/clusterapi_cluster_controller.go

```diff
@@ -0,0 +1,335 @@
+/*
```

Comment:
Filename here is a different convention than I've been doing (adding them in separate packages), but technically I think I like this better.